pyrobosim/test/utils/test_general_utils.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+from matplotlib.colors import CSS4_COLORS, to_rgb
+
+from pyrobosim.utils.general import parse_color
+
+
+def test_parse_color():
+    """Testing parse color with different input color formats"""
+    # Test with RGB list
+    color_rgb_list = (1.0, 0.0, 0.0)
+    assert parse_color([1.0, 0.0, 0.0]) == color_rgb_list
+
+    # Test with RGB tuple
+    color_rgb_tuple = (0.0, 1.0, 0.0)
+    assert parse_color((0.0, 1.0, 0.0)) == color_rgb_tuple
+
+    # Test with named color
+    color_by_name = "red"
+    assert parse_color("red") == to_rgb(CSS4_COLORS[color_by_name])
+
+    # Test with hexadecimal color format
+    color_hex = "#00FFFF"
+    assert parse_color("#00FFFF") == to_rgb(color_hex)
+
+    # Test with invalid RGB list
+    with pytest.raises(ValueError):
+        parse_color([1.0, 0.0])
```

Comment:
this is totally optional, but after these `with pytest.raises()` you can actually check the contents of the error too. For example:

```python
assert exc_info.value.args[0] == "Insufficient numbers."
```

Reply:
I will incorporate this suggestion in upcoming commits.